cert/spec.go

```diff
@@ -398,63 +365,89 @@ func (spec *Spec) ResetBackoff() {
 	spec.tr.Backoff.Reset()
 }
 
+// checkDiskCertKey performs sanity checks against the cert/key read from disk, identifying
+// if it's valid and still usable.
+func (spec *Spec) checkDiskCertKey(ca *x509.Certificate) error {
+	existingCert, err := spec.Cert.ReadCertificate()
+	if err != nil {
+		log.Debugf("spec %s: cert failed to be read: %s", spec, err)
+		return err
+	}
+	// update our internal time tracking while we're in here; even if immediately discard it,
+	// keeping it accurate to when we last saw it is desirable for metrics.
+	spec.updateCertExpiry(existingCert.NotAfter)
+
+	keyData, err := spec.Key.ReadFile()
+	if err != nil {
+		log.Debugf("spec %s: key failed to be read: %s", spec, err)
+		return err
+	}
+	err = verifyCertChain(ca, existingCert)
+	if err != nil {
+		log.Debugf("spec %s: CA has changed, cert is no longer valid via it: %s", spec, err)
```

Comment:
Maybe not here and this should be done at a more general level, but I think a cert changing on disk that results in validity changes should generate an info-level log message.

Reply:
same comment as the other- this ultimately gets fed into a log.info.  The debug logging at this layer is purely for ensuring there is a log line should the layer above make mistakes in handling.